core/src/main/java/org/apache/brooklyn/core/plan/PlanToSpecFactory.java

```diff
@@ -95,13 +96,18 @@ public static PlanToSpecTransformer forPlanType(ManagementContext mgmt, String p
         Collection<Exception> otherProblemsFromTransformers = new ArrayList<Exception>();
         for (PlanToSpecTransformer t: transformers) {
             try {
-                return Maybe.of(f.apply(t));
+                T result = f.apply(t);
+                if (result==null) {
+                    transformersWhoDontSupport.add(t.getShortDescription() + " (returned null)");
+                    continue;
+                }
+                return Maybe.of(result);
             } catch (PlanNotRecognizedException e) {
                 transformersWhoDontSupport.add(t.getShortDescription() +
                     (Strings.isNonBlank(e.getMessage()) ? " ("+e.getMessage()+")" : ""));
-            } catch (Exception e) {
+            } catch (Throwable e) {
```

Comment:
i think i was hoping to catch some weird errors, didn't realize propagateIfFatal would grab all `Error`.  maybe for some things like class linking etc we don't treat them as fatal?